### PR TITLE
regreet: add options for controlling theming

### PR DESCRIPTION
**Pull request by Enzime**
## Description of changes

Add options to match LightDM Greeters like `lightdm-gtk-greeter`: https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=lightdm.greeters+name

The defaults come from the default regreet configuration: https://github.com/rharish101/ReGreet/blob/main/regreet.sample.toml#L23-L33

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/programs/regreet.nix [hunk 1]
```diff
@@ -58,9 +58,100 @@ in
...
+
+    theme = {
+      package = lib.mkOption {
+        type = lib.types.package;
+        default = pkgs.gnome-themes-extra;
+        defaultText = lib.literalExpression "pkgs.gnome-themes-extra";
+        description = ''
+          The package that provides the theme given in the name option.
+        '';
+      };
```
Comment (fufexan) [inline nixos/modules/programs/regreet.nix:70]:
```suggestion
    theme = {
      package = lib.mkPackageOption pkgs "gnome-themes-extra" { } // {
        description = ''
          The package that provides the theme given in the name option.
        '';
      };
```
Package options should use `mkPackageOption`.

File: nixos/modules/programs/regreet.nix [hunk 2]
```diff
@@ -58,9 +58,100 @@ in
...
+        default = pkgs.cantarell-fonts;
+        defaultText = lib.literalExpression "pkgs.cantarell-fonts";
+        description = ''
+          The package that provides the font given in the name option.
+        '';
+      };
+
+      name = lib.mkOption {
+        type = lib.types.str;
+        default = "Cantarell 16";
```
Comment (fufexan) [inline nixos/modules/programs/regreet.nix:112]: Perhaps this should be 2 options, `name` and `size`?

**Review by fufexan — COMMENTED** [review]
I'm not sure whether it's necessary to add these options, as we already have `cfg.settings`, unlike lightdm's greeters.

**Comment by Enzime** [comment]
The value in adding these options is ensuring that these packages are installed, as I ended up spending some time today debugging why the icons in regreet and sway were broken for me

File: nixos/modules/programs/regreet.nix (hunk 1 above)
Comment (Enzime) [reply nixos/modules/programs/regreet.nix]: Fixed :+1

File: nixos/modules/programs/regreet.nix (hunk 2 above)
Comment (Enzime) [reply nixos/modules/programs/regreet.nix]: Fixed :+1:

**Review by fufexan — APPROVED** [review]
LGTM!

File: nixos/modules/programs/regreet.nix [hunk 3]
```diff
@@ -58,9 +58,96 @@ in
         modifiable properties.
       '';
     };
+
+    theme = {
```
Comment (SuperSandro2000) [inline nixos/modules/programs/regreet.nix:62]: Why are those not a subtype of the settings type?

**Pull request merged**